docs/content/latest/admin/yb-ts-cli.md

```diff
@@ -0,0 +1,208 @@
+---
+title: yb-ts-cli
+linkTitle: yb-ts-cli
+description: yb-ts-cli
+menu:
+  latest:
+    identifier: yb-ts-tool
+    parent: admin
+    weight: 2466
+isTocNested: 3
+showAsideToc: true
+---
+
+`yb-ts-cli` is a command line tool that can be used to perform an operation on tablet server (`yb-tserver`). Some of the commands perform operations similar to [`yb-admin` commands](../../../reference/configuration/yb-tserver).
+
+`yb-ts-cli` is a binary file installed with YugabyteDB and is located in the `bin` directory of the YugabyteDB home directory.
+
+## Syntax
+
+```sh
+yb-ts-cli [ --server_address=<addr> ] <command> <options>
+```
+
+- *addr*: The IP address of the tablet server. Default is `localhost`.
+- *command*: The operation to be performed. See [Commands](#commands) below.
+- *options*: The options to be applied to the command. See [Options](#options).
+
+## Command line help
+
+To display the available online help, run `yb-ts-cli` without any commands or options (flags) at the YugabyteDB home directory.
+
+```sh
+$ ./bin/yb-ts-cli
+```
+
+## Commands
+
+##### are_tablets_running
+
+If all tablets are running, returns "All tablets are running".
+
+**Syntax**
+
+```sh
+yb-ts-cli [ --server_address=<addr> ] are_tablets_running
+```
+
+##### count_intents
+
+Print the count of uncompleted intents (or [provisional records](../../../architecture/transactions/ditributed-txns/#provisional-records)). Useful for debugging and troubleshooting.
+
+**Syntax**
+
+```sh
+$ ./bin/yb-ts-cli  [ --server_address=<addr> ] count_intents
+```
+
+##### current_hybrid_time
+
+Prints the value of the current [hybrid time](../../../architecture/transactions/single-row-transactions/#hybrid-time-as-an-mvcc-timestamp).
+
+**Syntax**
+
+```sh
+$ ./bin/yb-ts-cli  [ --server_address=<addr> ] current_hybrid_time
+```
+
+##### delete_tablet
+
+Deletes the tablet with the specified tablet ID (`tablet_id`) and reason.
+
+**Syntax**
+
+```sh
+$ ./bin/yb-ts-cli  [ --server_address=<addr> ] delete_tablet <tablet_id> "<reason-string>"
+```
+
+- *tablet_id*: The identifier (ID) for the tablet.
+- *reason-string*: Text string noting why the tablet was deleted.
+
+##### dump_tablet
+
+Dump, or export, the specified tablet ID (`tablet_id`).
+
+**Syntax**
+
+```sh
+yb-ts-cli [ --server_address=<addr> ] dump_tablet <tablet_id>
+```
+
+- *tablet_id*: The identifier (ID) for the tablet.
+
+##### list_tablets
+
+Lists the tablets on the specified tablet server, displaying the following information:
+
+- Tablet id
+- State
+- Table name
+- Partition
+- Schema
+- properties
+  - lumn_name
+  - Tablet id
+  - State
+  - Table name
+  - Partition
+  - Schema
+
+**Syntax**
+
+```sh
+yb-ts-cli [ --server_address=<addr> ] list_tablets
+```
+
+##### set_flag
+
+Sets the specified configuration option (flag) for the tablet server.
+
+**Syntax**
+
+```sh
+$ ./bin/yb-ts-cli set_flag [ --force ] <flag> <value>
```

Comment:
This one is actually one of the most operationally useful commands and might warrant a bit more info.

This essentially changes the in-memory value of this flag, atomically, for a running process. This means that it can alter behavior of the respective process! It also means it does NOT persist across restarts!

In practice, there are some flags that are runtime safe to change and some that are not. These are generally denoted probably only in the code, for now. For example, the bind address of the server cannot be changed at runtime, since the server binds just once at startup.

One typical operational flow is that we can use this to modify runtime flags in memory and then out of band also modify the conf file that the server uses to come up. This allows for flags to be changed on processes, without executing a restart of the process!